call_EtherscanAPI.jl

```diff
@@ -1,33 +1,55 @@
 using HTTP, JSON
 
+# to override with your Etherscan API key
+API_token = ""
+# first block to start from
+first_block = 6290000
+log_result_topic = "0x8dd0b145385d04711e29558ceab40b456976a2b9a7d648cc1bcd416161bf97b9"
+contract_address = "0xA52e014B3f5Cc48287c2D483A3E026C32cc76E6d"
 
-API_token = "" # to override with your Etherscan API key
-fb = 6290000 #first block to start from
 
-log_array = [] #for stroing event logs from API call
-for i in 1:400 #the logs come in batches of 1000 — get as many as possible
-    global fb
-    global API_token
-    print(i,", ")
-    link = "https://api.etherscan.io/api?module=logs&action=getLogs&fromBlock=$fb&toBlock=latest&address=0xa52e014b3f5cc48287c2d483a3e026c32cc76e6d&topic0=0x8dd0b145385d04711e29558ceab40b456976a2b9a7d648cc1bcd416161bf97b9&apikey=$API_token"
-
-    r = HTTP.request("GET", link)
-    r = String(r.body)
-
-    x = JSON.parse(r)["result"]
-    push!(log_array, x)
-    fb = parse(Int64, x[end]["blockNumber"]) + 1 #for tracking progress
-    println(fb) 
-    
+# gets logs from a given block
+function get_logs(from_block; to_block="latest")
+    get_logs_url = "https://api.etherscan.io/api?module=logs&action=getLogs"
+    url = "$get_logs_url&fromBlock=$from_block&toBlock=$to_block&address=$contract_address&topic0=$log_result_topic&apikey=$API_token"
+    response = HTTP.request("GET", url)
+    body = String(response.body)
+    JSON.parse(body)["result"]
 end
 
+# pulls batches of logs and concatenates to a resulting array
+function get_logs_batch(from_block, batch_count)
+    # for stroing event logs from API call
+    log_array = []
+    for i in 1:batch_count
+        print(i,", ")
+        logs = get_logs(from_block)
+        # going to the next batch
+        from_block = parse(Int64, logs[end]["blockNumber"]) + 1
+        println(from_block)
+        log_array = vcat(log_array, logs)
```

Comment:
Note that we are now concatenating both arrays rather than appending.
https://docs.julialang.org/en/v1/base/arrays/#Base.vcat
This later eases iterating through it in `extract_oracle_results()` as the array is only one dimension

Reply:
Hi AndreMiras — awesome